flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/Dispatcher.java

```diff
@@ -428,6 +428,12 @@ private void stopDispatcherServices() throws Exception {
 
         try {
             if (isDuplicateJob(jobGraph.getJobID())) {
+                if (isGloballyTerminatedAndCleanedUp(jobGraph.getJobID())) {
+                    log.warn(
+                            "A job having the ID {} is already in globally-terminal state and marked as clean in the JobResultStore. No execution is going to happen.",
+                            jobGraph.getJobID());
```

Comment:
```suggestion
                    log.warn(
                            "Ignoring JobGraph submission '{}' ({}) because the job already reached a {} state in a previous execution.",
                            jobGraph.getName(), jobGraph.getJobID(), <insert job state>);
```